micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/response/executor/RestExecutor.groovy

```diff
@@ -80,25 +80,38 @@ final class RestExecutor<T> {
         }
     }
 
-    private ListenableFuture<ResponseEntity<T>> withRetry(HystrixCommand.Setter hystrix, Closure<ResponseEntity<T>> httpInvocation) {
+    private ListenableFuture<ResponseEntity<T>> withRetry(HystrixCommand.Setter hystrix, Closure<ResponseEntity<T>> hystrixFallback, Closure<ResponseEntity<T>> httpInvocation) {
         String correlationId = CorrelationIdHolder.get()
         return retryExecutor.getWithRetry {
             return CorrelationIdUpdater.withId(correlationId) {
-                return callHttp(hystrix, httpInvocation)
+                return callHttp(hystrix, hystrixFallback, httpInvocation)
             }
         }
     }
 
-    private ResponseEntity<T> callHttp(HystrixCommand.Setter hystrix, Closure<ResponseEntity<T>> httpInvocation) {
+    private ResponseEntity<T> callHttp(HystrixCommand.Setter hystrix, Closure<ResponseEntity<T>> hystrixFallback, Closure<ResponseEntity<T>> httpInvocation) {
         if(hystrix) {
-            return runInsideHystrixCommand(hystrix, httpInvocation)
+            return runInsideHystrixCommand(hystrix, hystrixFallback, httpInvocation)
         } else {
             return httpInvocation.call()
         }
     }
 
-    private ResponseEntity<T> runInsideHystrixCommand(HystrixCommand.Setter hystrix, Closure<ResponseEntity<T>> httpInvocation) {
+    private ResponseEntity<T> runInsideHystrixCommand(HystrixCommand.Setter hystrix, Closure<ResponseEntity<T>> hystrixFallback, Closure<ResponseEntity<T>> httpInvocation) {
         try {
+            if (hystrixFallback) {
+                return new CorrelatedCommand<ResponseEntity<T>>(hystrix) {
+                    @Override
+                    ResponseEntity<T> doRun() throws Exception {
+                        return httpInvocation.call()
+                    }
+
+                    @Override
+                    protected ResponseEntity<T> getFallback() {
+                        return hystrixFallback.call()
+                    }
+                }.execute()
+            }
             return new CorrelatedCommand<ResponseEntity<T>>(hystrix) {
```

Comment:
It's rather cosmetic one but could we use `else` here to highlight the second branch of if-else statement?